addons/calendar/models/calendar_event.py

```diff
@@ -322,10 +322,12 @@ def _compute_stop(self):
         # its recomputation. To avoid this we manually mark the field as computed.
         duration_field = self._fields['duration']
         self.env.remove_to_compute(duration_field, self)
-        for event in self.filtered('duration'):
+        for event in self:
             # Round the duration (in hours) to the minute to avoid weird situations where the event
             # stops at 4:19:59, later displayed as 4:19.
             event.stop = event.start + timedelta(minutes=round((event.duration or 1.0) * 60))
+            if not event.duration:
+                event.duration = 1.0
```

Comment:
This becomes really strange. Duration is a computed field that is 0 is no start / no stop; here we carefully avoid to recompute duration when computing stop do avoid circular computation. Then you re-set duration to 1.0

I think original purpose of duration is to stay at 0 if no start / stop. What are you trying to solve here ? Indeed there is already a "duration or 1.0" above.